nncf/torch/layer_utils.py

```diff
@@ -19,6 +24,33 @@
 COMPRESSION_MODULES = Registry("compression modules")
 
 
+class StatefullTorchModuleInterface(ABC):
+    """
+    Interface that should be implemented for every registered compression module to make it possible
+    to save an compression modules state and create an compression module from the saved state.
+    State of the module should be json serializable, no python objects except
+    standart (str, list and etc.) should be present in a compression module state.
+    The state for attributes with type torch.nn.Parameter
+    is recovered from the model `state_dict`, so there is no need to keep them in the module state.
+    Modules should avoid implementation of `__call__` method and use `forward` method instead,
+    as torch functions called inside the `__call__` method could not be unambiguously
+    separated from the wrapped parent nncf module functions calls, thus nncf is unable to
+    identify target point for that call during transformations recovery process.
+    """
+
+    @abstractmethod
+    def get_state(self) -> Dict[str, Any]:
```

Comment:
What do you think about `get_config`, `from_config` to align with high-level API?

Reply:
Done